### PR TITLE
update proptest state machine testing branch

### PR DESCRIPTION
**Pull request by tzemanovic**
The new proptest's branch includes initial state shrinking and some refactoring.

### CODE REVIEW

**Comment by juped** [comment]
actual build failure, branch name is wrong/

**Comment by juped** [comment]
should work now!

**Comment by tzemanovic** [comment]
> should work now!

thanks! I'll also set this to draft for now, there're couple issues

File: proof_of_stake/Cargo.toml [hunk 1]
```diff
@@ -13,5 +13,6 @@ borsh = "0.9.1"
 thiserror = "1.0.30"
 
 [dev-dependencies]
-proptest = {git = "https://github.com/heliaxdev/proptest", branch = "tomas/state-machine"}
+# A fork with state machime testing
```
Comment (juped) [inline proof_of_stake/Cargo.toml:16]: typo here, but i'll fix this before merging

File: wasm/wasm_source/Cargo.lock [hunk 2]
```diff
@@ -57,6 +57,7 @@ dependencies = [
  "hex",
  "ibc",
  "ibc-proto",
+ "ics23",
```
Comment (juped) [inline wasm/wasm_source/Cargo.lock:60]: unrelated?

**Review by juped — APPROVED** [review]

File: wasm/wasm_source/Cargo.lock (hunk 2 above)
Comment (juped) [reply wasm/wasm_source/Cargo.lock:60]: seems related

Comment (tzemanovic) [reply wasm/wasm_source/Cargo.lock:60]: I think this was perhaps missing on master
